itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/WarehouseInstance.java

```diff
@@ -379,6 +430,20 @@ public Table getTable(String dbName, String tableName) throws Exception {
     }
   }
 
+  /**
+   *
+   * @param dbName
+   * @param tableName
+   * @return the path of given table relative to the file system of ware house.
+   * @throws Exception
+   */
+  public String getTableLocation(String dbName, String tableName) throws Exception {
+    Table table = client.getTable(dbName, tableName);
+    String location = table.getSd().getLocation();
+    URI tabLocURI = (new Path(location)).toUri();
```

Comment:
There is not conversion involved, AFAIU. toURI() just returns uri set in Path. We need URI so that we can get the directory path from the location in storage descriptor which contains the whole URI including host, port etc.